qcfractal/storage_sockets/view.py

```diff
@@ -0,0 +1,134 @@
+import io
+import pathlib
+from typing import Any, Dict, Union
+
+import numpy as np
+import pandas as pd
+
+from qcelemental.util.serialization import serialize
+
+from ..interface.collections import HDF5View
+
+
+class ViewHandler:
+    def __init__(self, path: Union[str, pathlib.Path]) -> None:
+        """
+        Parameters
+        ----------
+        path: Union[str, Path]
+            Directory containing dataset views
+        """
+        self._path = pathlib.Path(path)
+        if not self._path.is_dir():
+            raise ValueError(f"Path in ViewHandler must be a directory, got: {self._path}")
+
+    def view_path(self, collection_id: int) -> pathlib.Path:
+        """
+        Returns the path to a view corresponding to a collection identified by an id.
+
+        Parameters
+        ----------
+        collection_id: int
+            Collection id corresponding to view
+
+        Returns
+        -------
+        pathlib.Path
+            Path of requested view
+        """
+
+        return self._path / f"{collection_id}.hdf5"
+
+    def view_exists(self, collection_id: int) -> bool:
+        """
+        Checks if view corresponding to a collection exists.
+
+        Parameters
+        ----------
+        collection_id: int
+            Collection id corresponding to view
+
+        Returns
+        -------
+        bool
+            Does the view exist?
+        """
+
+        return self.view_path(collection_id).is_file()
+
+    def handle_request(self, collection_id: int, request: str, model: Dict[str, Any]) -> Dict[str, Any]:
+        """
+        Handles REST requests related to views. This function implements the GET endpoint
+        /collections/[collection_id]/view/[request]
+
+        Parameters
+        ----------
+        collection_id: int
+            Collection id corresponding to a view.
+        request: str
+            Requested data. Allowed options and corresponding DatasetView methods:
+            - list: list_values
+            - value: get_values
+            - molecule: get_molecules
+            - entry: get_entries
+        model:
+            REST model containing input options.
+
+        Returns
+        -------
+        Dict[str, Any]:
+            Dictionary corresponding to requested REST model
+        """
+        meta = {"errors": [], "success": False, "error_description": False, "msgpacked_cols": []}
+
+        if not self.view_exists(collection_id):
+            meta["success"] = False
+            meta["error_description"] = f"View not available for collection #{collection_id}"
+            return {"meta": meta, "data": None}
+
+        view = HDF5View(self.view_path(collection_id))
```

Comment:
Is this pretty fast or should it be opened init?

Reply:
Should be very fast. The HDF5view constructor just sets `self._path = path`. The get methods in HDF5view open and close the HDF5 file every time they are called, which I benchmarked at 100 µs on my laptop. I believe this is the intended use of HDF5 (rather than keeping the file open throughout a session).